### PR TITLE
465 deprecated decorator

### LINKED ISSUE #465: @Deprecated decorator for controller endpoints
It would be nice to have a `@Deprecated` decorator to be able to update your swagger documentation to display endpoints that are not being used/maintained any more.

Right now you can accomplish this updating the `tsoa.json` file with the endpoints you want to deprecate.

### PR DESCRIPTION
**Pull request by fabripautasso**
## Background
I think it would be nice to have a decorator to mark deprecated endpoints

## Changes done
- Created new deprecated decorator function
- Updated methodGenerator
- Updated unit tests
- Updated documentation

**Closing issues**

Closes #465

### CODE REVIEW

File: tests/fixtures/controllers/methodController.ts [hunk 1]
```diff
@@ -85,11 +86,8 @@ export class MethodController extends Controller {
   public async oauthAndAPIkeySecurity(): Promise<TestModel> {
     return new ModelService().getModel();
   }
-
-  /**
-   * @deprecated
```
Comment (dgreene1) [inline tests/fixtures/controllers/methodController.ts]: Please revert the changes to this file since they test existing functionality that can't be removed.

File: src/metadataGeneration/methodGenerator.ts [hunk 2]
```diff
@@ -38,7 +38,7 @@ export class MethodGenerator {
     responses.push(this.getMethodSuccessResponse(type));
 
     return {
-      deprecated: isExistJSDocTag(this.node, tag => tag.tagName.text === 'deprecated'),
```
Comment (dgreene1) [inline src/metadataGeneration/methodGenerator.ts]: You can't remove the functionality that exists for users to already mark a method as deprecated via JSDocs. If you want this PR to be approved, you'll have to have `this.getIsDeprecated` check for the presence of a `@Deprecated` decorator or a JSDoc deprecated tag.

File: tests/fixtures/controllers/methodController.ts (hunk 1 above)
Comment (fabripautasso) [reply tests/fixtures/controllers/methodController.ts]: I just reverted these changes

File: src/metadataGeneration/methodGenerator.ts (hunk 2 above)
Comment (fabripautasso) [reply src/metadataGeneration/methodGenerator.ts]: You are right. I just modified `this.getIsDeprecated` to consider JSDoc deprecated tag

**Comment by fabripautasso** [comment]
@dgreene1 any other improvement/code change required for this PR?
